javaslang/src/test/java/javaslang/collection/LinkedHashMultimapTest.java

```diff
@@ -0,0 +1,131 @@
+package javaslang.collection;
+
+import javaslang.Tuple2;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+public class LinkedHashMultimapTest extends AbstractMultimapTest {
+
+    @Override
+    protected String className() {
+        return "Multimap[LinkedHashMap," + containerName() + "]";
+    }
+
+    @Override
+    protected <T1, T2> Multimap<T1, T2> emptyMap() {
+        switch (containerType) {
+            case SEQ:
+                return LinkedHashMultimap.withSeq().empty();
+            case SET:
+                return LinkedHashMultimap.withSet().empty();
+            case SORTED_SET:
+                return LinkedHashMultimap.withSortedSet(TreeSetTest.toStringComparator()).empty();
+        }
+        throw new RuntimeException();
```

Comment:
Better `throw new IllegalStateException("Unknown container type: " + containerType);`, also in all other places.